### PR TITLE
[WIP] radare2: 2.7.0 -> 2.8.0, radare2-cutter: 1.6 -> 1.7, refactor

### PR DESCRIPTION
**Pull request by dtzWill**
- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
   - [x] works-for-me ;)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
Now we have a problem that we check CS_TIP, which breaks `radare2-for-cutter`:

```
unpacking source archive /nix/store/7jx11qyfkdswvwhqnz20v6m703rk94sg-source
source root is source
patching sources
CS_TIP mismatch
builder for '/nix/store/99phyhxdssng0i5lb5vwmklk05nph36q-radare2-for-cutter-1.6.drv' failed with exit code 1
```

**Comment by dtzWill** [comment]
> It is probably the best to move radare2-cutter to the radare2 directory and let it have its own pinning.

hmm okay, agreed.  Sorry didn't see this earlier :).

**Comment by dtzWill** [comment]
Alright well this at least works and hopefully get things mostly to what you had in mind.  The update script needs fixing to work with radare2-cutter as well as radare2 (same info, new location).  Help appreciated but  no rush/pressure :).

File: pkgs/development/tools/analysis/radare2/default.nix [hunk 1]
```diff
@@ -15,66 +16,94 @@ assert pythonBindings -> python != null;
...
+  r2-for-cutter = generic {
+    version_commit = "19251";
+    gittap = "2.8.0-118-gb0547831f";
+    gittip = "b0547831f127b7357e3c93bc43933482a4d6213b";
+    rev = "b0547831f127b7357e3c93bc43933482a4d6213b";
+    version = "2018-08-07";
+    sha256 = "1ix42kipd1aayb494ajbxawzc1cwikm9fxk343d1kchxx4a30a1m";
+    cs_tip = "782ea67e17a391ca0d3faafdc365b335a1a8930a";
+    cs_sha256 = "1maww4ir78a193pm3f8lr2kdkizi7rywn68ffa65ipyr7j4pl6i4";
+  };
```
Comment (Mic92) [inline pkgs/development/tools/analysis/radare2/default.nix]: Please test

**Comment by dtzWill** [comment]
Looks great, thank you! \o/

File: pkgs/development/tools/analysis/radare2/update.py [hunk 2]
```diff
@@ -18,50 +20,107 @@ def sh(*args: str) -> str:
...
     url = "https://api.github.com/repos/radare/radare2/releases/latest"
     with urllib.request.urlopen(url) as response:
         release = json.load(response)  # type: ignore
-    version = release["tag_name"]
-    with tempfile.TemporaryDirectory() as dirname:
+    return release["tag_name"]
+
+
+def get_r2_cutter_rev() -> str:
+    url = "https://api.github.com/repos/radareorg/cutter/contents/"
```
Comment (Mic92) [inline pkgs/development/tools/analysis/radare2/update.py]: This should be the current version of cutter no the latest. I probably need to fix this in future.

**Pull request merged**